data/fixtures/recorded/surroundingPair/parseTree/python/clearPair5.yml

```diff
@@ -10,7 +10,7 @@ command:
           scopeType: {type: surroundingPair, delimiter: any}
   usePrePhraseSnapshot: true
 initialState:
-  documentContents: "def funk(words:list<str>):"
+  documentContents: "def funk(words:list[str]):"
```

Comment:
This is unfortunate. I'm guessing this is a test you added to make sure that pairs were predictable in the event of a bad parse. Seems the approach in this PR has backfired?

Reply:
We used to have logic to bail on parse-tree support if you're in an error node. Maybe we need to reintroduce that?

Reply:
Yes that was my intent, but we don't have access to the syntax node on the `QueryCapture` interface. With our logic of merging multiple captures I'm not even sure how to get the correct node.

Reply:
This is the last place we actually have access to the node on `match.captures`.
https://github.com/cursorless-dev/cursorless/blob/fb6b846efb96c595f16748ca5597b5e4ab441c0e/packages/cursorless-engine/src/languages/TreeSitterQuery/TreeSitterQuery.ts#L102

Reply:
I would just add an isError field on the final match and set it to true if any of the nodes in its captures have an error. Similar to how we use `some` form `allowMultiple` below

Reply:
Here is an implementation that restores the test by looking for error nodes. I'm not in love with how invasive this was, but I don't really know of a simpler solution.
https://github.com/cursorless-dev/cursorless/pull/2457/commits/709b19edb39e1937cc0dd0542c08a2327f9d065b

Reply:
Didn't we used to walk up ancestors looking for errors too? If so we should prob do that still

Reply:
This implementation is do in that

Reply:
> I would just add an isError field on the final match and set it to true if any of the nodes in its captures have an error. Similar to how we use `some` for `allowMultiple` below

The reason I'm not doing that is performance. Walking up the ancestor tree all the way to the route for every captured node in every scope scenario?

Reply:
We could of course add a method so we can do it lazily?

Reply:
This isn't to bad?
https://github.com/cursorless-dev/cursorless/pull/2457/commits/ec3a611f9506abef4ea9f177851bf2df0673f9d5